metricflow/engine/metricflow_engine.py

```diff
@@ -509,14 +510,15 @@ def simple_dimensions_for_metrics(self, metric_names: List[str]) -> List[Dimensi
             linkable_dimensions_tuple,
         ) in path_key_to_linkable_dimensions.items():
             for linkable_dimension in linkable_dimensions_tuple:
-                semantic_model = self._semantic_manifest_lookup.semantic_model_lookup.get_by_reference(
+                semantic_model = semantic_model = self._semantic_manifest_lookup.semantic_model_lookup.get_by_reference(
                     linkable_dimension.semantic_model_origin
                 )
                 assert semantic_model
                 dimensions.append(
                     Dimension.from_pydantic(
```

Comment:
Should we be renaming these `from_pydantic` methods to `from_semantic_model_<object>` or something?

Reply:
lets punt on that, I'm wondering if we could somehow clean these up and just pass the object directly?